ansible/playbooks/roles/upgrade/tasks/node-exporter.yml

```diff
@@ -18,30 +18,37 @@
     - k8s_as_cloud_service
   block:
     - name: Get information about helm releases
-      command: helm list --output json
+      command: helm list -A --output json
       register: helm_list
 
     - name: Check if node exporter release exist in default namespace and set fact
       set_fact:
         node_exporter_release_exists: >-
           {{ _names | ternary(true, false) }}
+        node_exporter_release_ns: >-
+          {{ _namespace }}
       vars:
         _names: >-
           {{ helm_list.stdout | from_json
                               | map(attribute='name')
                               | select('==', exporter_defaults.node_exporter_helm_chart_name)
                               | list }}
+        _namespace: >-
+          {{ helm_list.stdout | from_json
+                              | map(attribute='namespace')
+                              | list
+                              | first }}
```

Comment:
Is it ok here not to search for necessary namespace but just to take the first one? What's about removal `_names` var, getting `_namespace` the same way with select and then use `_namespace` in `node_exporter_release_exists`?

Reply:
ad 1) Seems to be efficient enough as the returned value is of list type and taking into consideration that we install node-exporter only in single ns.

ad2 )  We need to set both variables as the `_names` var ( so consequently `node_exporter_release_exists` ) allows us to find out the helm installation entity which is not possible when do the same but using `_namespace' var ( as You've proposed ) in the situation of upgrade combined with the ns move. It would only work in the scenario of just version upgrade.

Reply:
1. Have you tested if not only node-exported is installed by helm and there is a few releases?
2. If a namespace found, doesn't it mean that node-exporter release exists and vice versa?

Reply:
The remarks were taken on board within [this commit](https://github.com/epiphany-platform/epiphany/pull/2807/commits/5f575f515e7a519252a891e3585a0fc8ff329efe).